php/src/vaas/Vaas.php

```diff
@@ -81,15 +81,8 @@ public function forSha256Async(Sha256 $sha256, ?ForSha256Options $options = null
         return async(function () use ($sha256, $options, $cancellation) {
             $this->logger->debug("Requesting verdict for SHA256: $sha256");
 
-            if ($options === null) {
-                $options = new ForSha256Options(
-                    [
-                        'vaasRequestId' => null,
-                        'useHashLookup' => $this->options->useHashLookup ?? true,
-                        'useCache' => $this->options->useCache ?? true,
-                    ]
-                );
-            }
+            $options = $options ?? ForSha256Options::default();
```

Comment:
This will cause a bug. If the local and function specific options are null, we need to introduce new ForSha256Options with the global VaasOptions' settings injected to the local ones, if they are set, exactly like before. Just nullcheck and make default ForSha256Options like you changed it to, would overwrite VaasOptions and could cause VaaS not behave like a customer would like to. I suggest it like it was before but with your updated constructor:
```suggestion
            if ($options === null) {
                $options = new ForSha256Options($this->options->useCache, $this->options->useHashLookup, null);
            }
```